releasenotes/notes/deprecate-matmul-f6e2faa7f82439fb.yaml

```diff
@@ -0,0 +1,46 @@
+---
+features:
+  - |
+    Adds a ``&`` (``__and__``) binary operator to ``BaseOperator`` subclasses
+    in the :mod:`qiskit.quantum_info` module. This is shorthand to call the
+    classes ``compose`` method (``A & B == A.compose(B)``).
+
+    Example
+
+    .. code:: python
+    
+      import qiskit.quantum_info as qi
+
+      qi.Pauli('X') & qi.Pauli('Y')
+  - |
+    Adds a ``&`` (``__and__``) binary operator to
+    :class:`qiskit.quantum_info.Statevector` and
+    :class:`qiskit.quantum_info.DensityMatrix` classes. This is shorthand to
+    call the classes ``evolve`` method (``psi & U == psi.evolve(U)``).
+
+    Example
+
+    .. code:: python
+    
+      import qiskit.quantum_info as qi
+
+      qi.Statevector.from_label('0') & qi.Pauli('X')
+deprecations:
+  - |
+    Deprecates using ``@`` (``__matmul__``) for invoking the ``compose`` method
+    of ``BaseOperator`` subclasses. The ``compose`` method can instead be
+    invoked using the ``&`` (``__and__``) operator.
+    
+    In the future after the deprecation period is complete ``__matmul__`` will
```

Comment:
I'd remove this comment. Those are our plans but, in reality, we don't know yet.